state/src/interface/task_queue.rs

```diff
@@ -13,6 +14,15 @@ impl State {
     // | Getters |
     // -----------
 
+    /// Whether or not the task queue contains a specific task
+    pub fn contains_task(&self, task_id: &TaskIdentifier) -> Result<bool, StateError> {
+        let tx = self.db.new_read_tx()?;
+        let wallet_id = self.get_task_wallet(task_id)?;
```

Comment:
do we expect to have any tasks that are not associated with a wallet? if so, let's call out either in the doc comment or in a code comment that this method assumes all tasks are associated with a wallet, and its implementation would have to change if that assumption is changed

Reply:
Not right now no we don't